packages/fxa-settings/src/components/App/index.tsx

```diff
@@ -121,115 +128,125 @@ export const App = ({
-          {showReactApp && (
-            <>
-              <CannotCreateAccount path="/cannot_create_account/*" />
-              <Clear path="/clear/*" />
-              <CookiesDisabled path="/cookies_disabled/*" />
-
-              <Legal path="/legal/*" />
-              <LegalTerms path="/legal/terms/*" />
-              <LegalTerms path="/:locale/legal/terms/*" />
-              <LegalPrivacy path="/legal/privacy/*" />
-              <LegalPrivacy path="/:locale/legal/privacy/*" />
-
-              <ResetPassword path="/reset_password/*" />
-              <ConfirmResetPassword path="/confirm_reset_password/*" />
-
-              <WebChannelExample path="/web_channel_example/*" />
-
-              <LinkValidator
-                path="/complete_reset_password/*"
-                linkType={LinkType['reset-password']}
-                viewName="complete-reset-password"
-                getParamsFromModel={() => {
-                  return CreateCompleteResetPasswordLink();
-                }}
-              >
-                {({ setLinkStatus, params }) => (
-                  <CompleteResetPassword
-                    {...{
-                      setLinkStatus,
-                      params,
+      {/* ❌ TODO, create apollo client here or higher and remove from
+       * AppContext. In the meantime pull it off of AppContext and pass
+       * in here to keep it a singleton */}
+      {apolloClient && (
+        <ApolloProvider client={apolloClient}>
+          <Router basepath={'/'}>
+            <ScrollToTop default>
+              {/* We probably don't need a guard here with `showReactApp` or a feature flag/config
+               * check since users will be served the Backbone version of pages if either of those
+               * are false, but guard with query param anyway since we have it handy */}
+              {showReactApp && (
+                <>
+                  <CannotCreateAccount path="/cannot_create_account/*" />
+                  <Clear path="/clear/*" />
+                  <CookiesDisabled path="/cookies_disabled/*" />
+
+                  <Legal path="/legal/*" />
+                  <LegalTerms path="/legal/terms/*" />
+                  <LegalTerms path="/:locale/legal/terms/*" />
+                  <LegalPrivacy path="/legal/privacy/*" />
+                  <LegalPrivacy path="/:locale/legal/privacy/*" />
+
+                  <ResetPasswordContainer path="/reset_password/*" />
+                  <ConfirmResetPassword path="/confirm_reset_password/*" />
+
+                  <WebChannelExample path="/web_channel_example/*" />
+
+                  <LinkValidator
+                    path="/complete_reset_password/*"
+                    linkType={LinkType['reset-password']}
+                    viewName="complete-reset-password"
+                    getParamsFromModel={() => {
+                      return CreateCompleteResetPasswordLink();
```

Comment:
Random nit, that I just noticed, and that also came up in the oauth reset password work.

Inside `CreateCompleteResetPasswordLink` there is a call to useContext. Which it turns out can be problematic in some testing scenarios, because context maybe not be properly mocked and when looking at this react component's code it isn't obvious this call to useContext exists.

I think the purity of the function would be improved if we had the call `const { urlQueryData } = useContext(AppContext);` in the outer scope, and then just provided `urlQueryData` to `CreateCompleteResetPasswordLink`.